amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -155,14 +210,84 @@ describe('SearchPage', () => {
     });
 
     afterAll(() => {
-      getSelectedTabByResourceTypeSpy.mockRestore();
       createSearchOptionsSpy.mockRestore();
     });
   });
 
+  describe('componentDidUpdate', () => {
+    let props;
```

Comment:
To be a true unit test, we could make these improvements
1. Mock `getSanitizedUrlParams`
2. Mock `createSearchOptions`
3. Test the parameters that `searchAll` was called with.

This is a good practice even because even though the tests pass, this unit test has a dependency on the logic of the `getSanitizedUrlParams` and  `createSearchOptions`. When it comes to mocking, we mock/spyOn the methods we want to verify (like `setState` and `searchAll`), as well as the methods we want to treat as a black box (like `getSanitizedUrlParams` and  `createSearchOptions`).